messages.proto

```diff
@@ -31,50 +31,97 @@ message KillProcessRequest {
 message ExecutionStartingRequest {
+    /// Hold the current suite execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold the suite result in execution starting.
+    /// Some fields will not be populate before execution.
+    gauge.messages.ProtoSuiteResult suiteResult = 2;
 }
 
 /// Sent at end of Suite Execution. Tells the runner to execute `after_suite` hook.
 message ExecutionEndingRequest {
+    /// Hold the current suite execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold the suite result in execution ending.
+    gauge.messages.ProtoSuiteResult suiteResult = 2;
 }
 
 /// Sent at start of Spec Execution. Tells the runner to execute `before_spec` hook.
 message SpecExecutionStartingRequest {
+    /// Hold the current spec execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold the specs result in spec execution starting.
+    /// Some fields will not be populate before execution.
+    gauge.messages.ProtoSpecResult specResult = 2;
 }
 
 /// Sent at end of Spec Execution. Tells the runner to execute `after_spec` hook.
 message SpecExecutionEndingRequest {
+    /// Hold the current spec execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold the specs result in spec execution ending.
+    gauge.messages.ProtoSpecResult specResult = 2;
 }
 
 /// Sent at start of Scenario Execution. Tells the runner to execute `before_scenario` hook.
 message ScenarioExecutionStartingRequest {
+    /// Hold the current sceanrio execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold the scenarion result in scenarion execution starting.
+    /// Some fields will not be populate before execution.
+    gauge.messages.ProtoScenarioResult scenarioResult = 2;
 }
 
 /// Sent at end of Scenario Execution. Tells the runner to execute `after_scenario` hook.
 message ScenarioExecutionEndingRequest {
+    /// Hold the current scenario execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    gauge.messages.ProtoScenarioResult scenarioResult = 2;
 }
 
 /// Sent at start of Step Execution. Tells the runner to execute `before_step` hook.
 message StepExecutionStartingRequest {
+    /// Hold the current step execution info.
     ExecutionInfo currentExecutionInfo = 1;
+    /// Hold step result in step execution starting.
+    /// Some fields will not be populate before execution.
+    gauge.messages.ProtoStepResult stepResult = 2;
 }
 
 /// Sent at end of Step Execution. Tells the runner to execute `after_step` hook.
 message StepExecutionEndingRequest {
+    /// Hold the current step execution info.
     ExecutionInfo currentExecutionInfo = 1;
-}
-
-/// Contains details of the execution. 
+    /// Hold step result in step execution ending.
+    gauge.messages.ProtoStepResult stepResult = 2;
+}
+
+/// Contains command line arguments which passed by user during execution.
+message ExecutionArgs {
```

Comment:
This object can just hold the execution flags. Message can be
```
message ExecutionFlags {
  string flagName = 1;
  string flagValue = 2;
}
```
This kind of generic message ensures that all the flags are passed to the runner. If a new flag is added or removed, there need not be any worry about changing the proto message to include/exclude the field.